recipes/newrelic/infrastructure/otel/debian.yml

```diff
@@ -0,0 +1,295 @@
+          
+    
+    update_apt:
+      cmds:
+        - |
+          # Get latest definitions and skip any failure because of deprecation
+          OPTIONS="-o DPkg::Lock::Timeout=60 -o Acquire::Check-Valid-Until=false"
+          if [ ! -z "$HTTPS_PROXY" ]; then
+            OPTIONS="$OPTIONS -o Acquire::Http::Proxy={{.HTTPS_PROXY}}"
+          fi
+          apt-get $OPTIONS update -yq
+      silent: true
+      # apt will return an error if fails to update any of its sources. Ignore these errors and let the "install_infra" task fail.
+      ignore_error: true
+
+    install_gnupg:
+      cmds:
+        - |
+          if [ {{.HAS_GPG}} -eq 0 ] ; then
+            OPTIONS="-o DPkg::Lock::Timeout=60"
+            if [ ! -z "$HTTPS_PROXY" ]; then
+              OPTIONS="$OPTIONS -o Acquire::Http::Proxy={{.HTTPS_PROXY}}"
+            fi
+            apt-get $OPTIONS install gnupg2 -y
+          fi
+      vars:
+        HAS_GPG:
+          sh: command -v gpg | wc -l
+          
+    add_gpg_key:
+      cmds:
+        - |
+          curl -s {{.NEW_RELIC_DOWNLOAD_URL}}infrastructure_agent/keys/newrelic_apt_key_current.gpg | apt-key add -
+      silent: true
+
+    add_nr_source:
+      cmds:
+        - |
+          ARCH="amd64"
+          IS_ARM=$(uname -m | grep -i 'aarch64' | wc -l)
+          if [ $IS_ARM -gt 0 ] ; then
+            ARCH="arm64"
+          fi
+          if [ -n "{{.DEBIAN_CODENAME}}" ]; then
+            printf "deb [arch="$ARCH"] {{.NEW_RELIC_DOWNLOAD_URL}}infrastructure_agent/linux/apt {{.DEBIAN_CODENAME}} main" | tee /etc/apt/sources.list.d/newrelic-infra.list > /dev/null
+          else
+            printf "deb [arch="$ARCH"] {{.NEW_RELIC_DOWNLOAD_URL}}infrastructure_agent/linux/apt {{.DEBIAN_VERSION_CODENAME}} main" | tee /etc/apt/sources.list.d/newrelic-infra.list > /dev/null
+          fi
+      vars:
+        DEBIAN_CODENAME:
+          sh: awk -F= '/VERSION_CODENAME/ {print $2}' /etc/os-release
+        DEBIAN_VERSION_CODENAME:
+          sh: cat /etc/os-release | grep "VERSION=\"[0-9] " | awk -F " " '{print $2}' | sed 's/[()"]//g'
+      silent: true
+      
+    update_apt_nr_source:
+      cmds:
+        - |
+          # Get latest definitions and skip any failure because of deprecation
+          OPTIONS="-o DPkg::Lock::Timeout=60 -o Acquire::Check-Valid-Until=false"
+          if [ ! -z "$HTTPS_PROXY" ]; then
+            OPTIONS="$OPTIONS -o Acquire::Http::Proxy={{.HTTPS_PROXY}}"
+          fi
+          apt-get $OPTIONS update -yq
+      # apt will return an error if fails to update any of its sources. Ignore these errors and let the "install_infra" task fail.
+      ignore_error: true
+
+    install_otel_collector:
+      cmds:
+        - |
+          OPTIONS="-o DPkg::Lock::Timeout=60"
+          if [ ! -z "$HTTPS_PROXY" ]; then
+            OPTIONS="$OPTIONS -o Acquire::Http::Proxy={{.HTTPS_PROXY}}"
+          fi
+          apt-get $OPTIONS install nr-otel-collector -y -qq
```

Comment:
I'm seeing the same error testing with debian 10 and ubuntu 20.04 as I see in the failing unit test: https://github.com/newrelic/open-install-library/actions/runs/5022257087/jobs/9005521005?pr=921#step:7:3374

There might be an extra step we're missing here. @rogercoll do you happen to know why we're seeing `Unable to locate package nr-otel-collector`? Do we need to add an apt repo?